modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/value/ValueProviderFactory.java

```diff
@@ -71,10 +72,20 @@ ValueProvider createValueProvider() throws ValueResolvingException {
       injectValueProviderFields(resolver);
 
       if (factoryModelProperty.usesConnection()) {
+        Object connection = connectionSupplier.get();
```

Comment:
Create a task on CMTS side to also do this validation on MTS side. If the resolver defines that requires a configuration/connection we should check first if the parameterizableElementDeclaration has the configRef value set and the configRef is part of the session. This should also apply for metadataTypes resolution.